src/apps/renderer/hooks/antivirus/useAntivirus.tsx

```diff
@@ -0,0 +1,210 @@
+import { useEffect, useState } from 'react';
+
+export interface SelectedItemToScanProps {
+  path: string;
+  itemName: string;
+  isDirectory: boolean;
+}
+
+export type ScanType = 'files' | 'folders';
+export interface ScannedItemsProps {
+  file: string;
+  isInfected: boolean | null;
+  viruses: string[];
+}
+export type Views = 'locked' | 'chooseItems' | 'scan';
+
+export interface UseAntivirusReturn {
+  infectedFiles: string[];
+  currentScanPath?: string;
+  countScannedFiles: number;
+  view: Views;
+  isScanning: boolean;
+  isScanCompleted: boolean;
+  progressRatio: number;
+  isAntivirusAvailable: boolean;
+  isDefenderActive: boolean;
+  onScanUserSystemButtonClicked: () => Promise<void>;
+  onScanAgainButtonClicked: () => void;
+  onCancelScan: () => void;
+  onCustomScanButtonClicked: (scanType: ScanType) => Promise<void>;
+  onRemoveInfectedItems: (infectedFiles: string[]) => Promise<void>;
+  isWinDefenderActive: () => Promise<boolean>;
+}
+
+export const useAntivirus = (): UseAntivirusReturn => {
+  const [infectedFiles, setInfectedFiles] = useState<string[]>([]);
+  const [currentScanPath, setCurrentScanPath] = useState<string>();
+  const [countScannedFiles, setCountScannedFiles] = useState<number>(0);
+  const [progressRatio, setProgressRatio] = useState<number>(0);
+  const [isScanCompleted, setIsScanCompleted] = useState(false);
+  const [isScanning, setIsScanning] = useState(false);
+  const [isAntivirusAvailable, setIsAntivirusAvailable] = useState<boolean>(false);
+  const [isDefenderActive, setIsDefenderActive] = useState<boolean>(false);
+  const [view, setView] = useState<Views>('locked');
+
+  useEffect(() => {
+    window.electron.antivirus.onScanProgress(handleProgress);
+    return () => {
+      window.electron.antivirus.removeScanProgressListener();
+    };
+  }, []);
+
+  useEffect(() => {
+    isUserElegible();
+    isWinDefenderActive();
+  }, []);
+
+  const isUserElegible = async () => {
+    try {
+      const isAntivirusAvailable = await window.electron.antivirus.isAvailable();
+
+      if (!isAntivirusAvailable) {
+        setView('locked');
+        return;
+      }
+
+      setIsAntivirusAvailable(true);
+      setView('chooseItems');
+    } catch (error) {
+      //
+      setIsAntivirusAvailable(false);
+      setView('locked');
+    }
+  };
+
+  const isWinDefenderActive = async () => {
+    try {
+      const isWinDefenderActive = await window.electron.antivirus.isDefenderActive();
+
+      setIsDefenderActive(isWinDefenderActive);
+      return isWinDefenderActive;
+    } catch (error) {
```

Comment:
No error handling here, is it intentional? If so, just use `finally` instead of `catch`

Reply:
In case it is an error I use an specific state for it